src/jit/gentree.cpp

```diff
@@ -17355,29 +17355,38 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoad()
     {
         return true;
     }
-    else if (category == HW_Category_IMM)
+    else if (HWIntrinsicInfo::MaybeMemoryLoad(gtHWIntrinsicId))
     {
         // Some AVX instructions here also have MemoryLoad sematics
```

Comment:
While you're here, "sematics" should be "semantics"

Reply:
Oops, will fix